### PR TITLE
Making mount options aware

### LINKED ISSUE #1795: mount resource remounts on every chef-client run
As discussed in  #1781 this is probably undesirable.

### PR DESCRIPTION
**Pull request by JamesBelchamber**
Fixes #1795

### CODE REVIEW

**Comment by thommay** [comment]
hey @mattray , could y'all lend some AIX help to the test cases here?

File: spec/unit/provider/mount/mount_spec.rb [hunk 1]
```diff
@@ -317,9 +317,16 @@
         @provider.mount_fs()
       end
 
-      it "should not mount the filesystem if it is mounted" do
-        allow(@current_resource).to receive(:mounted).and_return(true)
-        expect(@provider).not_to receive(:shell_out!)
+      it "should remount the filesystem if it is mounted and the options have changed" do
+        options = "rw,noexec,noauto"
+        @new_resource.options(%w{rw exec auto})
+        @provider.should_receive(:shell_out!).with("mount -t ext3 -o remount,rw,exec,auto /dev/sdz1 /tmp/foo")
```
Comment (siddheshwar-more) [inline spec/unit/provider/mount/mount_spec.rb]: @JamesBelchamber 
This expectation is invalid here as @provider.mount_fs() never calls remount_fs. Its a def action_mount who calls action_remount where remount_fs gets invoked. 
Could you please move this test to  spec/unit/provider/mount_spec.rb ?

**Comment by siddheshwar-more** [comment]
@btm @mattray 
PR to Foxed AIX mount spec -  https://github.com/JamesBelchamber/chef/pull/1

File: spec/unit/provider/mount/windows_spec.rb [hunk 2]
```diff
@@ -106,9 +106,17 @@ class Volume
         @provider.mount_fs
       end
 
-      it "should not mount the filesystem if it is mounted" do
-        expect(@vol).not_to receive(:add)
-        allow(@current_resource).to receive(:mounted).and_return(true)
+      it "should remount the filesystem if it is mounted and the options have changed" do
+        @vol.should_receive(:add).with(:remote => @new_resource.device,
+                                       :username => @new_resource.username,
```
Comment (siddheshwar-more) [inline spec/unit/provider/mount/windows_spec.rb]: @JamesBelchamber 
To make travis happy could you please replace @vol.should_receive with following
expect(@vol)..to receive(:add).with(....). As `should_receive` is deprecated in latest rspec version.

File: spec/unit/provider/mount/windows_spec.rb [hunk 3]
```diff
@@ -106,9 +106,17 @@ class Volume
...
+                                       :domainname => @new_resource.domain,
+                                       :password => @new_resource.password)
+        @provider.mount_fs
+      end
+
+      it "should not mount the filesystem if it is mounted and the options have not changed" do
+        @vol.should_not_receive(:add)
```
Comment (siddheshwar-more) [inline spec/unit/provider/mount/windows_spec.rb]: Same here use: `expect(@vol).to_not receive(:add)`

File: spec/unit/provider/mount/windows_spec.rb [hunk 4]
```diff
@@ -106,9 +106,17 @@ class Volume
...
+        @current_resource.stub(:mounted).and_return(true)
```
Comment (siddheshwar-more) [inline spec/unit/provider/mount/windows_spec.rb]: Replace above with `allow(@current_resource).to receive(:mounted).and_return(true)`

**Comment by JamesBelchamber** [comment]
Why is this marked as 'waiting contributor response'? Pretty sure everything is ready to go, after @siddheshwar-more made his changes :)

**Comment by thommay** [comment]
:+1: ping @chef/client-core

**Comment by tyler-ball** [comment]
:+1:

**Comment by lamont-granquist** [comment]
:+1:

**Pull request closed without merging**